test/Dialect/FIRRTL/inferRW.mlir

```diff
@@ -0,0 +1,235 @@
+      firrtl.connect %4, %index : !firrtl.uint<4>, !firrtl.uint<4>
+      %5 = firrtl.subfield %mem_0_MPORT_1(1) : (!firrtl.bundle<addr: uint<4>, en: uint<1>, clk: clock, data: uint<8>, mask: uint<1>>) -> !firrtl.uint<1>
+      firrtl.connect %5, %wen : !firrtl.uint<1>, !firrtl.uint<1>
+      %6 = firrtl.subfield %mem_0_MPORT_1(2) : (!firrtl.bundle<addr: uint<4>, en: uint<1>, clk: clock, data: uint<8>, mask: uint<1>>) -> !firrtl.clock
+      firrtl.connect %6, %clock : !firrtl.clock, !firrtl.clock
+      %7 = firrtl.subfield %mem_0_MPORT_1(3) : (!firrtl.bundle<addr: uint<4>, en: uint<1>, clk: clock, data: uint<8>, mask: uint<1>>) -> !firrtl.uint<8>
+      firrtl.connect %7, %data_0 : !firrtl.uint<8>, !firrtl.uint<8>
+      %8 = firrtl.subfield %mem_0_MPORT_1(4) : (!firrtl.bundle<addr: uint<4>, en: uint<1>, clk: clock, data: uint<8>, mask: uint<1>>) -> !firrtl.uint<1>
+      firrtl.connect %8, %_T_29 : !firrtl.uint<1>, !firrtl.uint<1>
+      %9 = firrtl.not %wen : (!firrtl.uint<1>) -> !firrtl.uint<1>
+      firrtl.connect %mem_MPORT_en, %9 : !firrtl.uint<1>, !firrtl.uint<1>
+      %REG = firrtl.reg %clock  : !firrtl.uint<1>
+      firrtl.connect %REG, %9 : !firrtl.uint<1>, !firrtl.uint<1>
+      %r_0 = firrtl.reg %clock  : !firrtl.uint<8>
+      %10 = firrtl.mux(%REG, %mem_MPORT_data_0, %r_0) : (!firrtl.uint<1>, !firrtl.uint<8>, !firrtl.uint<8>) -> !firrtl.uint<8>
+      firrtl.connect %r_0, %10 : !firrtl.uint<8>, !firrtl.uint<8>
+      %11 = firrtl.mux(%REG, %mem_MPORT_data_0, %r_0) : (!firrtl.uint<1>, !firrtl.uint<8>, !firrtl.uint<8>) -> !firrtl.uint<8>
+      firrtl.connect %auto_0, %11 : !firrtl.uint<8>, !firrtl.uint<8>
+
+     // CHECK:  %[[v7:.+]] = firrtl.mux(%[[writeEnable:.+]], %[[writeAddr:.+]], %[[readAddr:.+]]) : (!firrtl.uint<1>, !firrtl.uint<4>, !firrtl.uint<4>) -> !firrtl.uint<4>
+     // CHECK:  firrtl.connect %[[v0:.+]], %[[v7]] : !firrtl.uint<4>, !firrtl.uint<4>
+     // CHECK:  %[[v8:.+]] = firrtl.or %[[readEnable:.+]], %[[writeEnable]] : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+     // CHECK:  firrtl.connect %[[v1:.+]], %[[v8]] : !firrtl.uint<1>, !firrtl.uint<1>
+     // CHECK:  firrtl.connect %[[v4:.+]], %[[writeEnable]] : !firrtl.uint<1>, !firrtl.uint<1>
+     // CHECK:  %9 = firrtl.wire  : !firrtl.clock
+     // CHECK:  firrtl.connect %[[readAddr]], %[[index2:.+]] : !firrtl.uint<4>, !firrtl.uint<4>
+     // CHECK:  firrtl.connect %[[readEnable]], %mem_MPORT_en : !firrtl.uint<1>, !firrtl.uint<1>
+     // CHECK:  firrtl.connect %[[writeAddr]], %index : !firrtl.uint<4>, !firrtl.uint<4>
+     // CHECK:  firrtl.connect %[[writeEnable]], %wen : !firrtl.uint<1>, !firrtl.uint<1>
+     // CHECK:  %[[v10:.+]] = firrtl.not %wen : (!firrtl.uint<1>) -> !firrtl.uint<1>
+     // CHECK:  firrtl.connect %mem_MPORT_en, %[[v10]] : !firrtl.uint<1>, !firrtl.uint<1>
+    }
+}
+
+firrtl.circuit "sram6t"   {
```

Comment:
Can you add some comments above each test as to what this is checking?

Supreme nit: what does `sram6t` mean? Hardware folks will read this and think "6 transitor SRAM" which I don't think is what this is?